src/com/ms/silverking/cloud/dht/collection/IntArrayCuckoo.java

```diff
@@ -7,401 +7,398 @@
+      int baseOffset;
+
+      //displayEntry(bucketIndex, entryIndex);
+      baseOffset = getHTEntryIndex(bucketIndex, entryIndex);
+      //System.out.println("\t"+ entryIndex +"\t"+ bucketIndex +"\t"+ baseOffset);
+      //System.out.printf("%x:%x\t%x:%x\n",
+      //        buf[baseOffset + mslOffset], buf[baseOffset + lslOffset],
+      //        msl, lsl);
+      return buf[baseOffset + mslOffset] == msl && buf[baseOffset + lslOffset] == lsl;
+    }
+
+    int getValue(int bucketIndex, int entryIndex) {
+      return values[bucketIndex * entriesPerBucket + entryIndex];
+    }
+
+    long getMSL(int bucketIndex, int entryIndex) {
+      int baseOffset;
+
+      baseOffset = getHTEntryIndex(bucketIndex, entryIndex);
+      return buf[baseOffset + mslOffset];
+    }
+
+    long getLSL(int bucketIndex, int entryIndex) {
+      int baseOffset;
+
+      baseOffset = getHTEntryIndex(bucketIndex, entryIndex);
+      return buf[baseOffset + lslOffset];
+    }
+
+    private void putValue(int index, long msl, long lsl, int value, int entryIndex) {
+      int baseOffset;
+
+      //System.out.println(index +"\t"+ bucketIndex);
+      assert entryIndex < entriesPerBucket;
+      baseOffset = getHTEntryIndex(index, entryIndex);
+      buf[baseOffset + mslOffset] = msl;
+      //System.out.println("baseOffset: "+ baseOffset +"\tmsl: "+ buf[entrySize * index + mslOffset));
+      buf[baseOffset + lslOffset] = lsl;
+      values[index * entriesPerBucket + entryIndex] = value;
+    }
+  }
+
+  @Override
+  public Iterator<DHTKeyIntEntry> iterator() {
+    return new CuckooIterator();
+  }
+
+  class CuckooIterator extends CuckooIteratorBase implements Iterator<DHTKeyIntEntry> {
+    CuckooIterator() {
+      super();
+    }
+
+    @Override
+    public DHTKeyIntEntry next() {
+      DHTKeyIntEntry mapEntry;
+
+      // precondition: moveToNonEmpty() has been called
+      mapEntry = new DHTKeyIntEntry(subTables[subTable].getMSL(bucket, entry),
+          subTables[subTable].getLSL(bucket, entry), subTables[subTable].getValue(bucket, entry));
+      moveToNonEmpty();
+      return mapEntry;
+    }
+
+    boolean curIsEmpty() {
+      return subTables[subTable].getValue(bucket, entry) == empty;
+    }
+  }
+
+  public static IntArrayCuckoo rehash(IntArrayCuckoo oldTable) {
+    IntArrayCuckoo newTable;
+
+    newTable = new IntArrayCuckoo(oldTable.getConfig().doubleEntries());
+    try {
+      for (DHTKeyIntEntry entry : oldTable) {
+        //System.out.println(entry);
+        newTable.put(entry.getKey(), entry.getValue());
+      }
+      return newTable;
+    } catch (TableFullException tfe) {
+      throw new RuntimeException("Unexpected table full during rehash");
+    }
+  }
+
+  public static IntArrayCuckoo rehashAndAdd(IntArrayCuckoo oldTable, DHTKey key, int value) {
+    IntArrayCuckoo newTable;
+
+    newTable = rehash(oldTable);
+    try {
+      newTable.put(key, value);
+      return newTable;
+    } catch (TableFullException tfe) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)